src/esn.calendar.libs/app/services/calendar-service.js

```diff
@@ -371,5 +375,21 @@ require('./calendar-cache.js');
         FileSaver.saveAs(calendarBlob, `${calendarId}.ics`);
       });
     }
+    /**
+      * Get the generated token for the secret link
+      * @param {Object} jwtPayload The jwtPayload that contains the information to generate token
+      * @return {String} a generated token with calendarHomeId and calendar id payload
+      */
+    function generateTokenForSecretLink(jwtPayload) {
+      return $http({ method: 'POST', url: getGenerateJWTforSecretLinkUrl(), data: jwtPayload });
```

Comment:
1. We don't handle sending requests in `services/calendar-service.js`. It should behandled it `services/calendar-api.js`.
2. We should be using `calendarRestangular` wherever possible instead of `$http` because we have some configurations for it.